portal/lib/template_menu.php

```diff
@@ -10,110 +10,50 @@
-        $tpls = get_template_dir_array($root_directory . $pid_path, $pid_path);
-        if (count($tpls) > 0) {
-            $dir_list[$gen_const] = $tpls;
-        }
-    }
-    // get only directories from our category list
-    foreach ($category_list as $cat) {
-        if (stripos($cat['option_id'], 'repository') !== false) {
-            continue;
-        }
-        if (substr($root_directory, -1) !== "/") {
-            $root_directory .= "/";
-        }
-        if ($cat_dir_iter = get_template_dir_array($root_directory . convert_safe_file_dir_name($cat['option_id']), $cat['option_id'])) {
-            $dir_list[$cat['title']] = $cat_dir_iter;
-        }
-    }
 
-    return $dir_list;
-}
+render_template_list($pid, $cuser);
 
-function render_template_list($tree)
+function render_template_list($pid, $cuser)
 {
-    global $pid, $cuser;
-
-    foreach ($tree as $key => $file) {
+    $templateService = new DocumentTemplateService();
+    $category_list = $templateService->getFormattedCategories();
+    $templates = $templateService->getTemplateListAllCategories();
+    $patient_templates = $templateService->getTemplateCategoriesByPatient();
+    $templates = array_merge_recursive($templates, $patient_templates);
+    foreach ($templates as $key => $file) {
         if (is_array($file)) {
-            $is_category = $key;
-
+            $is_category = $category_list[$key]['title'] ?? $key;
+            if ($is_category == 'Default') {
+                $is_category = '';
+            }
             $cat_name = text($is_category);
-            echo "<li class='text-center'><h5 class='mb-0'>$cat_name</h5></li>\n";
+
+            $flag = false;
             foreach ($file as $filename) {
-                if (is_array($filename)) {
+                if ($filename['template_name'] == 'Help') {
                     continue;
                 }
-                $basefile = basename($filename, ".tpl");
-                $btnname = text(ucwords(str_replace('_', ' ', $basefile)));
-                $btnfile = attr($filename);
-                echo '<li class="nav-item mb-1"><a class="nav-link text-success btn btn-outline-success" id="' . $basefile . '"' . ' href="#" onclick="page.newDocument(' . "$pid,'$cuser','$btnfile')" . '"' . ">$btnname</a></li>\n";
+                if ((int)$filename['pid'] !== 0 && (int)$filename['pid'] !== (int)$pid) {
+                    continue;
+                }
+                if (!$flag) {
+                    $flag = true;
+                    echo "<li class='text-center'><h5 class='mb-0'>$cat_name</h5></li>\n";
+                }
+                $id = attr($filename['id']);
+                $btnname = text($filename['template_name']);
+                echo '<li class="nav-item mb-1 template-item"><a class="nav-link text-success btn btn-sm btn-outline-success" id="' . $id . '"' . ' href="#" onclick="page.newDocument(' . "'$pid','$cuser','$btnname', '$id')" . '"' . ">$btnname</a></li>\n";
```

Comment:
I would change the above 3 lines to following to clarify the escaping (since gotta use attr_js on $id and $btnname):
```
                $id = $filename['id'];
                $btnname = $filename['template_name'];
                echo '<li class="nav-item mb-1 template-item"><a class="nav-link text-success btn btn-sm btn-outline-success" id="' . attr($id) . '"' . ' href="#" onclick="page.newDocument(' . attr_js($pid) . ', ' . attr_js($cuser) . ', ' . attr_js($btnname) . ', ' . attr_js($id) . ')">' . text($btnname) , "</a></li>\n";
```